server/src/streaming/batching/batch_accumulator.rs

```diff
@@ -71,35 +71,32 @@ impl BatchAccumulator {
     pub fn materialize_batch_and_maybe_update_state(&mut self) -> (bool, RetainedMessageBatch) {
         let batch_base_offset = self.base_offset;
         let batch_last_offset_delta = (self.current_offset - self.base_offset) as u32;
-        let batch_max_timestamp = self.messages.last().unwrap().timestamp;
         let split_point = std::cmp::min(self.capacity as usize, self.messages.len());
-        let (batch, remainder) = self.messages.as_slice().split_at(split_point);
+
         let mut bytes = BytesMut::with_capacity(self.current_size.as_bytes_u64() as usize);
-        for message in batch {
+        let batch: Vec<_> = self.messages.drain(..split_point).collect();
```

Comment:
you technically, don't have to collect this there, but looping over this will cause the entire iterator to be consumed so it's w/e

Reply:
yeah, nice catch, thank you! 
done